### PR TITLE
add ucsc_bed12tobigbed module

### PR DESCRIPTION
**Pull request by yuukiiwa**
Testing the integration of modules from [nanoseq dsl2 pull request](https://github.com/nf-core/nanoseq/pull/103) to nf-core/modules

### CODE REVIEW

File: software/ucsc/bed12tobigbed/main.nf [hunk 1]
```diff
@@ -0,0 +1,31 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+def options    = initOptions(params.options)
+
+process UCSC_BED12TOBIGBED {
+    tag "$sample"
```
Comment (drpatelh) [inline software/ucsc/bed12tobigbed/main.nf:8]:
```suggestion
    tag "$meta.id"
```

File: software/ucsc/bed12tobigbed/main.nf [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    label 'process_medium'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:'ucsc', publish_id:'') }
```
Comment (drpatelh) [inline software/ucsc/bed12tobigbed/main.nf:12]:
```suggestion
        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:meta.id) }
```

File: software/ucsc/bed12tobigbed/main.nf [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    conda     (params.enable_conda ? "bioconda::ucsc-bedtobigbed=377" : null)
+    container "quay.io/biocontainers/ucsc-bedtobigbed:377--h446ed27_1"
+    if ({ task.exitStatus in [255] }) { errorStrategy = 'ignore' }
```
Comment (drpatelh) [inline software/ucsc/bed12tobigbed/main.nf:16]:
```suggestion
    conda (params.enable_conda ? "bioconda::ucsc-bedtobigbed=377" : null)
    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
        container "https://depot.galaxyproject.org/singularity/ucsc-bedtobigbed:377--h446ed27_1"
    } else {
        container "quay.io/biocontainers/ucsc-bedtobigbed:377--h446ed27_1"
    }
```

File: software/ucsc/bed12tobigbed/main.nf [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+    input:
+    tuple val(sample), path(sizes), path(bed12)
+
+    output:
+    tuple val(sample), path(sizes), path("*.bigBed"), emit: bigbed
```
Comment (drpatelh) [inline software/ucsc/bed12tobigbed/main.nf:22]:
```suggestion
    input:
    tuple val(meta), path(bed)
    path  sizes

    output:
    tuple val(meta), path("*.bigBed"), emit: bigbed
    path "*.version.txt"             , emit: version
```

File: software/ucsc/bed12tobigbed/main.nf [hunk 5]
```diff
@@ -0,0 +1,31 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+def options    = initOptions(params.options)
+
```
Comment (drpatelh) [inline software/ucsc/bed12tobigbed/main.nf:6]:
```suggestion

def VERSION = '377'

```

File: software/ucsc/bed12tobigbed/main.nf [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    output:
+    tuple val(sample), path(sizes), path("*.bigBed"), emit: bigbed
+
+    script:
+    """
+    bedToBigBed \\
+        $bed12 \\
+        $sizes \\
+        ${sample}.bigBed
+    """
```
Comment (drpatelh) [inline software/ucsc/bed12tobigbed/main.nf:30]:
```suggestion
    script:
    def software = getSoftwareName(task.process)
    def prefix   = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
    """
    bedToBigBed \\
        $bed \\
        $sizes \\
        ${prefix}.bigBed

    echo $VERSION > ${software}.version.txt
    """
```

File: tests/data/bed/test_ucsc_bed12tobigbed.bed [hunk 7]
```diff
@@ -0,0 +1,15 @@
+chr7	113726364	114271760	uc003vgt.2	0	+	113726364	113726364	0	12	302,145,91,178,98,211,90,58,138,51,219,178,	0,76634,201709,340192,376764,412573,448307,452016,542230,543282,543477,545218,
+chr7	113726364	114333827	uc003vgu.3	0	+	113726364	113726364	0	20	302,91,178,90,58,75,138,51,201,178,214,105,88,84,77,102,122,70,164,3991,	0,201709,340192,448307,452016,484497,542230,543282,543495,545218,556100,558375,565893,567616,573046,573262,575755,577140,577963,603472,
```
Comment (drpatelh) [inline tests/data/bed/test_ucsc_bed12tobigbed.bed:2]: Can we rename this file to `tests/data/bed/test.bed12`

File: tests/data/chrom_size/hg19.chrom.sizes [hunk 8]
```diff
@@ -0,0 +1,93 @@
+chr1	249250621
+chr2	243199373
+chr3	198022430
+chr4	191154276
+chr5	180915260
+chr6	171115067
+chr7	159138663
```
Comment (drpatelh) [inline tests/data/chrom_size/hg19.chrom.sizes:7]: We just need chromosome 7 for the tests in this file.

File: tests/data/chrom_size/hg19.chrom.sizes [hunk 9]
```diff
@@ -0,0 +1,93 @@
+chr1	249250621
+chr2	243199373
```
Comment (drpatelh) [inline tests/data/chrom_size/hg19.chrom.sizes:2]: Can we rename this file to `tests/data/sizes/test.sizes` ?

File: tests/data/chrom_size/hg19.chrom.sizes (hunk 8 above)
Comment (drpatelh) [reply tests/data/chrom_size/hg19.chrom.sizes:7]: Ideally, we need to keep the tests down to a minimal size as possible.

**Review by drpatelh — CHANGES_REQUESTED** [review]
Looks great overall @yuukiiwa However, the format of the modules we submit here will require a few modifications compared to what we have in nanoseq to be able to add them. Have a look at the docs in the main README or maybe the example module I [wrote](https://github.com/nf-core/modules/blob/master/software/TOOL/SUBTOOL/main.nf) for future PRs. 

How do you feel about getting a full list of modules together for nanoseq in an issue there and we can discuss whether whether we can add them here.I suspect we won't be able to add all of them because of the way we are using one genome per sample, and that won't be appropriate for many samples.

File: tests/data/bed/test_ucsc_bed12tobigbed.bed (hunk 7 above)
Comment (yuukiiwa) [reply tests/data/bed/test_ucsc_bed12tobigbed.bed]: I have changed the name to tests/data/bed/test.bed12

File: tests/data/chrom_size/hg19.chrom.sizes (hunk 8 above)
Comment (yuukiiwa) [reply tests/data/chrom_size/hg19.chrom.sizes]: Only chr7 remains now.

File: tests/data/chrom_size/hg19.chrom.sizes (hunk 9 above)
Comment (yuukiiwa) [reply tests/data/chrom_size/hg19.chrom.sizes]: I have changed the name to tests/data/sizes/test.sizes

**Comment by yuukiiwa** [comment]
Hi @drpatelh, Sorry for the delayed reply! Thank you for your suggestions! I have modified them accordingly.

**Comment by yuukiiwa** [comment]
New pull request is here:https://github.com/nf-core/modules/pull/302

**Pull request closed without merging**